docs/DeveloperGuide.md

```diff
@@ -1,9 +1,123 @@
 
+### Architecture
+
+![Architecture Diagram](https://github.com/DeetoMok/tp/raw/d14fb309c9f816a46ecd19ed26f74b8c8ac1ac03/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the Module Manager Application.
+
+Module Manager consists of a main class called Duke responsible for
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other
+* At shut down: Shuts down the components and invokes cleanup method where necessary
+
+The other components involved are:
+
+`UI`: The user interface of the application
+
+`Storage`: Reads data from, and writes data to, the hard disk
+
+#### UI component
+The `UI` component consists of a `Ui` class that stores all user interaction output data. 
+It displays all user interactions to the user based on the command line inputs received.
+
+The `UI` component,
+
+*   Executes user commands using the `Logic` component
+
+#### Logic component
+The `Logic` component 
+
+1. `Logic` uses the `Parser` class to parse the user command.
+2. `Parser` then returns a specific command class e.g. `AddCommand`, `FindCommand` etc. 
+which is executed by the main class `Duke`.
+All these command classes inherits from the abstract `Command` class.
+3. The command execution can affect the Model (e.g. adding a module in ModuleList)
+4. The result of the command execution is passed back to the Ui.
+5. In addition, the command execution can also instruct the `Ui` to perform certain actions, 
+such as displaying help to the user.
+
+#### Model component
+The `Model` component
+* Stores a `Person` object that represents the user's details. (e.g. `totalCap`, `totalModuleCreditCompleted`)
+* Stores the `ModuleList` and `SemesterList` Data
+* Does not depend on any of the other three components
+
+#### Storage component
+
+
+
+### Implementation
+
+#### `Add to Semester` feature 
+The `Add to Semester` mechanism is facilitated by `AddtoSemCommand` which extends from `Command`. 
+It allows `ModuleManager` to assign a module to a semester by adding the module to a 
+`SemModulesList`, it implements the following operations:
+
+Given below is an example how the `Add to Semester` behaves at each step.
+
+##### Step 1:
+The user launches the application for the first time. The `SemesterList` will be initialized with the none 
+`SemModulesList` in it.
+
+##### Step 2:
+When users enter a add to semester command, e.g `add id/CS2113 s/4 mc/4`, this command will be parsed in `Parser`
+and then `Parser` returns a `AddToSemCommand`. `AddToSemCommand` then calls `Command#excuted(SemesterList semesterList,
+ AvailableModulesList availableModulesList) `
+`(AddToSemCommand#excuted(SemesterList semesterList, AvailableModulesList availableModulesList))`
+
+##### Step 3:
+`AddToSemCommand#excuted()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
+ then calls `AddToSemCommand#checkModuleExist(semesterList)` to check whether the selected 
```

Comment:
The "checkModuleExitst()" method is not reflected in the sequence diagram. Would it be more comprehensible to add it in?

Reply:
Okay, thanks!